packages/h3/src/H3Adapter.ts

```diff
@@ -0,0 +1,169 @@
+
+    return this;
+  }
+
+  public setViewsPath(viewPath: string): H3Adapter {
+    this.viewPath = viewPath;
+
+    return this;
+  }
+
+  public setErrorHandler(handler: (error: Error) => ControllerHandlerReturnType) {
+    this.errorHandler = handler;
+
+    return this;
+  }
+
+  public setApiRoutes(routes: AppControllerRoute[]): H3Adapter {
+    this.apiRoutes = routes;
+
+    return this;
+  }
+
+  public setEntryRoute(routeDef: AppViewRoute): H3Adapter {
+    this.entryRoute = routeDef;
+
+    return this;
+  }
+
+  public setQueues(bullBoardQueues: BullBoardQueues): H3Adapter {
+    this.bullBoardQueues = bullBoardQueues;
+    return this;
+  }
+
+  public setUIConfig(config: UIConfig = {}): H3Adapter {
+    this.uiConfig = config;
+
+    return this;
+  }
+
+  public registerHandlers() {
+    if (!this.statics) {
+      throw new Error(`Please call 'setStaticPath' before using 'registerPlugin'`);
+    } else if (!this.entryRoute) {
+      throw new Error(`Please call 'setEntryRoute' before using 'registerPlugin'`);
+    } else if (!this.viewPath) {
+      throw new Error(`Please call 'setViewsPath' before using 'registerPlugin'`);
+    } else if (!this.apiRoutes) {
+      throw new Error(`Please call 'setApiRoutes' before using 'registerPlugin'`);
+    } else if (!this.bullBoardQueues) {
+      throw new Error(`Please call 'setQueues' before using 'registerPlugin'`);
+    } else if (!this.errorHandler) {
+      throw new Error(`Please call 'setErrorHandler' before using 'registerPlugin'`);
+    }
+
+    const getStaticPath = (relativePath: string) => {
+      if (!this.statics) return '';
+
+      return `${this.statics.path}${relativePath.replace(
+        `${this.basePath}${this.statics.route}`,
+        ''
+      )}`;
+    };
+
+    const { method, route, handler } = this.entryRoute;
+
+    const routes = Array.isArray(route) ? route : [route];
+
+    routes.forEach((route) => {
+      this.uiHandler.use(
+        `${this.basePath}${route}`,
+        eventHandler(async () => {
+          const { name: filename, params } = handler({
+            basePath: this.basePath,
+            uiConfig: this.uiConfig,
+          });
+
+          return ejs.renderFile(`${this.viewPath}/${filename}`, params);
+        }),
+        method
+      );
+    });
+
+    this.uiHandler.get(
+      `${this.basePath}${this.statics.route}/**`,
+      eventHandler(async (event) => {
+        await serveStatic(event, {
+          fallthrough: true,
+          indexNames: undefined,
+          getContents: (id) => readFileSync(getStaticPath(id)),
+          getMeta: (id) => {
+            const fileStat = statSync(getStaticPath(id));
+
+            return {
+              size: fileStat.size,
+              type: getContentType(id),
+            };
+          },
+        });
+      })
+    );
+
+    this.apiRoutes.forEach(({ route, handler, method }) => {
+      this.uiHandler.use(
+        `${this.basePath}${route}`,
```

Comment:
`route` has an interface of `string | string[]`, you should make sure it is an array, and run for each route.

One way to do so is using `[].concat(route)` which will convert it to an array of strings, then you can run for each on of the values.